include/zcbor_common.h

```diff
@@ -281,8 +281,8 @@ bool zcbor_union_end_code(zcbor_state_t *state);
  *  payload, payload_len, and elem_count are used to initialize the first state.
  *  in the array, which is the state that can be passed to cbor functions.
  */
-void zcbor_new_state(zcbor_state_t *state_array, uint_fast32_t n_states,
-		const uint8_t *payload, size_t payload_len, uint_fast32_t elem_count);
+void zcbor_new_state(zcbor_state_t *state_array, size_t n_states,
+		const uint8_t *payload, size_t payload_len, size_t elem_count);
 
 #ifdef ZCBOR_STOP_ON_ERROR
 /** Check stored error and fail if present, but only if stop_on_error is true. */
```

Comment:
Other task should be probably created for this, but IMO it would be nice to have a little piece of documentation describing what `zcbor_check_error()` returns in case of an error.